### ISSUE TITLE
Remove the `cirq` dependency

### ISSUE THREAD

**Issue by nelimee** [QCHackers/tqec#333]
**Is your feature request related to a problem? Please describe.**
`cirq` has been a dependency of `tqec` since day 1. It is currently used as an intermediate format: we build the circuit representing the QEC computation in a `cirq.Circuit` and, when the circuit is complete, we translate it to a `stim.Circuit`.

Initially, this intermediate representation was justified by the ease of manipulation provided by `cirq` when compared to `stim`: it is easier to mutate a `cirq.Circuit` than a `stim.Circuit`. 

But the more I think about it, the more I think we do not really need the ease of manipulation brought by `cirq.Circuit`.

Additionally:

- `cirq` is an heavy dependency that has a lot of sub-dependencies we do not really require but still have to install,
- the translation from `cirq.Circuit` to `stim.Circuit` is not negligible in terms of runtime (see #329),
- as long as we do not change too much out circuits, I *expect* (without any benchmark to support that claim) `stim.Circuit` to be faster than `cirq.Circuit`. 

**Describe the solution you'd like**
Being able to remove `cirq` from the base `tqec` dependencies. It can still appear as an *optional* dependency, but default users should not need to install `cirq` to use `tqec`.

**Describe alternatives you've considered**
Keep the `cirq` dependency. But because we are currently in a refactoring period, I think this is the right time to try out things.

**Additional context**
I will submit a tentative PR and explore this issue. If it turns out we can remove the `cirq` dependency without too much problems, then this is nice. If solving this issue is too hard, then maybe `cirq` is really needed after all.

**Comment by inmzhang** [comment]
I can work on the noise model part since it's kind of independent from the others. I'll bring code from [Craig's implementation](https://github.com/Strilanc/inplace-y-basis-2023/blob/main/src/midout/gen/_noise.py) so it should not be much work.

**Comment by nelimee** [comment]
> I can work on the noise model part since it's kind of independent from the others. I'll bring code from [Craig's implementation](https://github.com/Strilanc/inplace-y-basis-2023/blob/main/src/midout/gen/_noise.py) so it should not be much work.

I started to look into it yesterday, it seems to be working perfectly. One warning on licence: I found the code in CC BY 4.0 on Zenodo, but there is no licence on the Github version. That means we should use the Zenodo version, and explicitly write down the licence somewhere (I made that in the module comment).

I will push what I have done today, I have a nearly working implementation without cirq. There are still some bugs, but most of the code should be working. That will be a huge PR though, sorry for that.
